doc/internals.rst

```diff
@@ -42,6 +42,37 @@ xarray objects via the (readonly) :py:attr:`Dataset.variables
 <xarray.Dataset.variables>` and
 :py:attr:`DataArray.variable <xarray.DataArray.variable>` attributes.
 
+Duck arrays
+-----------
+
+.. warning::
+
+    This is a experimental feature.
+
+xarray can wrap custom `duck array`_ objects as long as they define numpy's
+`shape`, `dtype` and `ndim` properties and the `__array__`, `__array_ufunc__`
+and `__array_function__` methods.
+
+In certain situations (e.g. when printing the variables of ``Dataset``), xarray
+will display the repr of a `duck array`_ in a single line, limiting it to a
+certain number of characters. If that would drop too much information, the
+`duck array`_ may define a ``_repr_inline_`` method:
```

Comment:
```suggestion
`duck array`_ may define a ``_repr_inline_`` method that takes `max_width` (number of characters) as an argument:
```

Reply:
looks good, except that information is also part of the docstring of the example / skeleton. Should I remove the docstring and use this instead?

Reply:
I think the redundancy is OK. I don't feel too strongly about this change so if you don't like it, feel free to ignore

Reply:
Agree it's good to have it in the prose
I think fine either way re repeating in docstring or not

Reply:
I'll keep both